packages/editor/src/components/dialogs/CreatePrefabPanelDialog.tsx

```diff
@@ -0,0 +1,124 @@
+Version 1.0. (the "License"); you may not use this file except in compliance
+with the License. You may obtain a copy of the License at
+https://github.com/EtherealEngine/etherealengine/blob/dev/LICENSE.
+The License is based on the Mozilla Public License Version 1.1, but Sections 14
+and 15 have been added to cover use of software over a computer network and 
+provide for limited attribution for the Original Developer. In addition, 
+Exhibit A has been modified to be consistent with Exhibit B.
+
+Software distributed under the License is distributed on an "AS IS" basis,
+WITHOUT WARRANTY OF ANY KIND, either express or implied. See the License for the
+specific language governing rights and limitations under the License.
+
+The Original Code is Ethereal Engine.
+
+The Original Developer is the Initial Developer. The Initial Developer of the
+Original Code is the Ethereal Engine team.
+
+All portions of the code written by the Ethereal Engine team are Copyright © 2021-2023 
+Ethereal Engine. All Rights Reserved.
+*/
+
+import { PopoverState } from '@etherealengine/client-core/src/common/services/PopoverState'
+import { staticResourcePath } from '@etherealengine/common/src/schema.type.module'
+import { Engine, Entity, hasComponent, setComponent } from '@etherealengine/ecs'
+import { ModelComponent } from '@etherealengine/engine/src/scene/components/ModelComponent'
+import { getState } from '@etherealengine/hyperflux'
+import Button from '@etherealengine/ui/src/primitives/tailwind/Button'
+import Input from '@etherealengine/ui/src/primitives/tailwind/Input'
+import Modal from '@etherealengine/ui/src/primitives/tailwind/Modal'
+import React, { useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { EditorControlFunctions } from '../../functions/EditorControlFunctions'
+import { exportRelativeGLTF } from '../../functions/exportGLTF'
+import { EditorState } from '../../services/EditorServices'
+import { HeirarchyTreeNodeType } from '../hierarchy/HeirarchyTreeWalker'
+
+export default function CreatePrefabPanel({ node }: { node?: HeirarchyTreeNodeType }) {
+  const entity = node?.entity as Entity
+  const [defaultPrefabFolder, setDefaultPrefabFolder] = useState<string>('/assets/custom-prefabs')
+  const [prefabName, setPrefabName] = useState<string>('prefab')
+  const [prefabTag, setPrefabTag] = useState<string[]>([])
+  const { t } = useTranslation()
+
+  const onExportPrefab = async () => {
+    if (!hasComponent(entity, ModelComponent)) {
+      EditorControlFunctions.addOrRemoveComponent([entity], ModelComponent, true)
+      setComponent(entity, ModelComponent)
+    }
+    const editorState = getState(EditorState)
+    const fileName = defaultPrefabFolder + '/' + prefabName + '.gltf'
+    const srcProject = editorState.projectName!
+    try {
+      exportRelativeGLTF(entity, srcProject, fileName)
+      //pass static resource
+      const resources = await Engine.instance.api.service(staticResourcePath).find({
```

Comment:
this will always fail because the file doesn't exist yet

Reply:
we should remove this call, and construct a static resource object to pass for upload